### DISCUSSION TITLE
Popup support for dependency injection

### DISCUSSION THREAD

**Discussion by Vroomer** [CommunityToolkit/Maui#405] [New Feature Discussions]
I'm trying to implement popup with ViewModel that has dependance on several services. Unfortunately `public static void ShowPopup<TPopup>(this Page page, TPopup popup)` doesn't utilize Maui dependency injection. I thought about resolving the Popup with ServiceProvider first, but that brings a problem of providing the Popup constructor with anchor parameter. It would be great if ShowPopup method could be provided with type of Popup and resolve the Popup using Maui dependency injection on its own, including option to provide anchor as a parameter. If you know other way to achieve this, I will be glad for you suggestions.

**Comment by pictos** [comment]
You can resolve the services by creating some extension method to `GetServices` and use it. We didn't integrate our popup with DI, maybe we can add this in the future.
If you need reference on how you can create your extension method, check [this](https://github.com/davidortinau/WeatherTwentyOne/blob/main/src/WeatherTwentyOne/Services/ServiceExtensions.cs)

**Reply by tcerdaj** [reply]
Hi team, for ShowPopup there is a way to add animation parameter as Rg.Plugin.Popup does it?

I am doing the popup animation when the page is loading, but I'd like to do it when the ShowPoup method is invoked. In that way, I can use the popup for my custom toast, and other crazy design implementations.

Also when I set the position to end to the Toolkit popup is moving from top to bottom, I will like that the popup start showed from the specified position.

This property BackgroundInputTransparent="True" is also important to show/hide the background.

Thanks,

Teofilo Cerda.

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
